### PR TITLE
[ALS-5034] [ALS-5125] Add customizations for GIC named dataset save, view & manage.

### PR DESCRIPTION
**Pull request by srpiatt**

### CODE REVIEW

File: custom-migrations/auth/V20__ADD_SAVE_DATASET_ROLE.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
+use auth;
+
+SET @uuidRule = REPLACE(UUID(),'-','');
+INSERT INTO access_rule (
```
Comment (Luke-Sikina) [inline custom-migrations/auth/V20__ADD_SAVE_DATASET_ROLE.sql:4]: This formatting is kinda gross. I respect matching columns to their values, but I think you can do more than 1 per line

File: custom-migrations/auth/V20__ADD_SAVE_DATASET_ROLE.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+VALUES (
+    unhex(@uuidPriv),
+    unhex(@uuidRule)
+);
+
+INSERT INTO role_privilege ( role_id, privilege_id )
+VALUES (
+    (SELECT uuid FROM role WHERE name = 'PIC-SURE Top Admin'),
+    unhex(@uuidPriv)
+)
```
Comment (srpiatt) [inline custom-migrations/auth/V20__ADD_SAVE_DATASET_ROLE.sql:55]: Can remove admins -- all admins are users.

File: custom-migrations/auth/V20__ADD_SAVE_DATASET_ROLE.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+INSERT INTO role_privilege ( role_id, privilege_id )
+VALUES (
+    (SELECT uuid FROM role WHERE name = 'PIC-SURE User'),
+    unhex(@uuidPriv)
+)
```
Comment (srpiatt) [inline custom-migrations/auth/V20__ADD_SAVE_DATASET_ROLE.sql:60]: Close up these lines~

File: ui/src/main/webapp/picsureui/overrides/dataset/dataset-save.hbs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<div id="named-dataset">
+    <section id="dataset-query-info">
+        <p>Fill out the following information to save your dataset ID for future use. To view and manage your dataset ID's, go to the "User Profile" tab.</p>
+        <div id="dataset-name-container" class="row">
+            <div class="title">Dataset ID Name:</div>
+            <div class="content"><input type="text" id="dataset-name" value="" class="tabable" /></div>
```
Comment (srpiatt) [inline ui/src/main/webapp/picsureui/overrides/dataset/dataset-save.hbs:6]: Maybe remove this file.

**Pull request merged**